cmd/rqlite/backup.go

```diff
@@ -33,3 +61,82 @@ func backup(ctx *cli.Context, filename string, argv *argT) error {
 	ctx.String("backup file written successfully\n")
 	return nil
 }
+
+func makeRestoreRequest(restoreFile io.Reader) func(string) (*http.Request, error) {
+	return func(urlStr string) (*http.Request, error) {
+		req, err := http.NewRequest("POST", urlStr, restoreFile)
+		req.Header["Content-type"] = []string{"text/plain"}
+		if err != nil {
+			return nil, err
+		}
+		return req, nil
+	}
+}
+
+func restore(ctx *cli.Context, filename string, argv *argT) error {
+	statusURL := fmt.Sprintf("%s://%s:%d/status", argv.Protocol, argv.Host, argv.Port)
+	client := http.Client{Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: argv.Insecure},
+	}}
+	statusResp, err := client.Get(statusURL)
+	if err != nil {
+		return err
+	}
+	defer statusResp.Body.Close()
+
+	if statusResp.StatusCode == http.StatusUnauthorized {
+		return fmt.Errorf("unauthorized")
+	}
+
+	body, err := ioutil.ReadAll(statusResp.Body)
+	if err != nil {
+		return err
+	}
+	statusRet := &statusResponse{}
+	if err := parseResponse(&body, &statusRet); err != nil {
+		return err
+	}
+	if statusRet.Store == nil {
+		return fmt.Errorf("unexpected server response: store status not found")
+	}
+
+	restoreFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	// It is cheaper to append the actual pragma command to the restore file
+	isFkConstraintEnabledInStore := statusRet.Store.SqliteStatus.FkConstraint == "enabled"
```

Comment:
Nit: this variable name is unwieldy. Go style dictates shorter variable names, especially since it's just used over as few lines. Please use a shorter name, like `fkEnabled`.